### PR TITLE
Add number precision error messages

### PR DESCRIPTION
**Pull request by colinrotherham**
This PR closes [bug #414012](https://dev.azure.com/defragovuk/DEFRA-CDP/_workitems/edit/414012) and adds error messages for:

* `number.precision`
* `number.integer`
* `number.unsafe`

## Whole numbers
When number fields are set to `precision: 0` we validate whole numbers only

Number integer error message

## Decimal numbers
When number fields are set to `precision: 2` etc we validate maximum decimal places

Number precision error message

### CODE REVIEW

File: src/server/plugins/engine/components/NumberField.ts [hunk 1]
```diff
@@ -23,7 +23,11 @@ export class NumberField extends FormComponent {
 
     const { options, schema, title } = def
 
-    let formSchema = joi.number().label(title.toLowerCase()).required()
+    let formSchema = joi
+      .number()
+      .custom(getValidatorPrecision(this))
```
Comment (colinrotherham) [inline src/server/plugins/engine/components/NumberField.ts:28]: Preventing rounding using `.prefs({ convert: false })` made every "string number" invalid 😔

I've used a custom validator instead so we can keep string-to-number conversion

File: src/server/plugins/engine/components/NumberField.ts [hunk 2]
```diff
@@ -111,3 +117,31 @@ export class NumberField extends FormComponent {
     return typeof value === 'number'
   }
 }
+
+export function getValidatorPrecision(component: NumberField) {
```
Comment (mokhld) [inline src/server/plugins/engine/components/NumberField.ts:121]: Prob being pedantic here (sorry!), but this func is returning a validator so maybe `getPrecisionValidator` is more apt? Initially reading this I thought the function returns a precision value.

[the remark this answers is not in the document]
Comment (mokhld) [reply src/server/plugins/engine/components/NumberField.ts:121]: Sorry ignore me, saw that you've typed it already below 😅

**Review by mokhld — APPROVED** [review]

File: src/server/plugins/engine/components/NumberField.ts (hunk 2 above)
Comment (colinrotherham) [reply src/server/plugins/engine/components/NumberField.ts:121]: Yeah it makes sense now 😆

We have `getValidatorXXX` elsewhere but if you fancy changing things like this in another PR?

E.g. I'm slowly tackling confusing names in this one for example:

* https://github.com/DEFRA/forms-runner/pull/515

**Pull request merged**